src/mqtt_client.erl

```diff
@@ -148,7 +148,10 @@ start_options(Opts) ->
     }.
 
 start_options_transport(KeepAlive, {Type, Opts}) ->
-    Host = maps:get(host, Opts, "localhost"),
```

Comment:
Actually I think this should be moved even below into `mqtt_transport`.

Today we have
`mqtt_transport:host() :: inet:ip_address() | inet:hostname().`

If we extend it to be
`mqtt_transport:host() :: inet:ip_address() | inet:hostname() | binary().`

Then we will just need to add a clause where we handle the binary case in `mqtt_transport:open/4` and `mqtt_transport:open/5`.